eos/statistics/log-likelihood.cc

```diff
@@ -972,106 +993,94 @@ namespace eos
             }
 
             // compute cholesky decomposition of covariance matrix
-            gsl_matrix * cholesky()
+            void cholesky()
             {
-                // dimensionality of parameter space
-                unsigned k = mean.size();
-
                 // copy covariance matrix
-                gsl_matrix * L = gsl_matrix_alloc (k, k);
-                for (unsigned i = 0 ; i < k ; ++i)
-                {
-                    for (unsigned j = 0 ; j < k ; ++j)
-                        gsl_matrix_set(L, i, j, covariance[i][j]);
-                }
+                gsl_matrix_memcpy(_chol, _covariance);
+                gsl_linalg_cholesky_decomp(_chol);
+            }
 
-                gsl_linalg_cholesky_decomp(L);
+            // invert covariance matrix based on previously obtained Cholesky decomposition
+            void invert_covariance()
+            {
+                // copy cholesky matrix
+                gsl_matrix_memcpy(_covariance_inv, _chol);
 
-                return L;
+                // compute inverse matrix from cholesky
+                gsl_linalg_cholesky_invert(_covariance_inv);
             }
 
+
             virtual LogLikelihoodBlockPtr clone(ObservableCache cache) const
             {
-                std::array<ObservableCache::Id, n_> indices;
+                auto k = _mean->size;
```

Comment:
Copying your nitpicking on @akokulu7 PR: why don't you declar `k const`?

Reply:
Done.